core/src/main/scala/sttp/tapir/server/interceptor/decodefailure/DecodeFailureHandler.scala

```diff
@@ -156,60 +156,12 @@ object DefaultDecodeFailureHandler {
       val base = failureSourceMessage(ctx.failingInput)
 
       val detail = ctx.failure match {
-        case InvalidValue(errors) if errors.nonEmpty => Some(ValidationMessages.validationErrorsMessage(errors))
-        case Error(_, error: JsonDecodeException)    => Some(error.getMessage)
-        case _                                       => None
+        case InvalidValue(i: ValidationResult.Invalid[_]) => Some(i.description)
+        case Error(_, error: JsonDecodeException)         => Some(error.getMessage)
+        case _                                            => None
       }
 
       combineSourceAndDetail(base, detail)
     }
   }
-
-  /** Default messages when the decode failure is due to a validation error. */
-  object ValidationMessages {
```

Comment:
The goal here was to separate the strucutre of the validators from rendering them as a string value. What's the benefit of moving this logic inside `ValidatorResult` trait?